### ISSUE TITLE
Missing Test Case - 3020. Find the Maximum Number of Elements in Subset

### ISSUE THREAD

**Issue by rohan-v8rma** [LeetCode-Feedback/LeetCode-Feedback#19803]
### LeetCode Username
rohan-v8rma

### Problem number, title, and link
1. Find the Maximum Number of Elements in Subset https://leetcode.com/problems/find-the-maximum-number-of-elements-in-subset/

### Category of the bug
- [ ] Problem description
- [ ] Solved examples
- [ ] Problem constraints
- [ ] Problem hints
- [ ] Incorrect or missing "Related Topics"
- [ ] Incorrect test Case *(Output of test case is incorrect as per the problem statement)*
- [x] Missing test Case *(Incorrect/Inefficient Code getting accepted because of missing test cases)*
- [ ] Editorial
- [ ] Programming language support

### Language used for code

Java

### Code used for Submit/Run operation

```java
class Solution {
    static int getCount(int num, int[] ct) {
        int actualCt = (int)Math.min(ct[1], 1);
        
        int baseVal = ct[0];
        
        if(num == baseVal) {
            return actualCt;
        }
        
        num = (int)Math.sqrt(num);
        
        while(num != baseVal) {
            actualCt += 2;
            num = (int)Math.sqrt(num);
        }
        
        actualCt += 2;
        
        return actualCt;
    }
    
    public int maximumLength(int[] nums) {
        Arrays.sort(nums);
        
        int i = 0;
        
        while(i < nums.length && nums[i] == 1) {
            i++;
        }
        
        int maxCt = i - (1 - (i % 2)); // Odd number of 1s can be in subset.
        
        HashMap<Integer, int[]> subsetMap = new HashMap<>();
        
        for(; i < nums.length; i++) {
            int[] elementDetails = subsetMap.getOrDefault(nums[i], new int[2]);
            
            int countOfElement = elementDetails[1];

            if(countOfElement > 0) {
                elementDetails[1]++;
                continue;
            }
            
            double sqrtDouble = Math.sqrt(nums[i]);
            
            if(sqrtDouble % 1 == 0) {
                  int sqrt = (int)sqrtDouble;

                  elementDetails = subsetMap.getOrDefault(sqrt, new int[2]);
                  
                  int countOfSqrt = elementDetails[1];

                  if(countOfSqrt >= 1) {
                      subsetMap.remove(sqrt);
                      if(countOfSqrt == 1) {
                          // Since sqrt only seen 1 time, subset ends here.
                          maxCt = Math.max(getCount(sqrt, elementDetails), maxCt);    
                      }
                      else {
                          // 1 "nums[i]" seen. all smaller numbers seen twice or more
                          elementDetails[1] = 1; 

                          subsetMap.put(nums[i], elementDetails);    
                      }         
	              continue; 
                  }
            }
            
            // nums[i] is the starting of a new subset.
            elementDetails[0] = nums[i];
            elementDetails[1] = 1; // First nums[i] seen.
            subsetMap.put(nums[i], elementDetails);
        }
    
        for(Map.Entry<Integer, int[]> ct: subsetMap.entrySet()) {
            maxCt = Math.max(getCount(ct.getKey(), ct.getValue()), maxCt);
        }
        
        return maxCt;
    }
}
```

### Expected behavior

- In this array, the expected maximum length subset is [4, 16, 4], whose length is 3.
- However, the submitted code returns 1. 

- This is because the `continue` statement is placed like this:
  ```java
  if(countOfSqrt >= 1) {
    subsetMap.remove(sqrt);
    if(countOfSqrt == 1) {
      // Since sqrt only seen 1 time, subset ends here.
      maxCt = Math.max(getCount(sqrt, elementDetails), maxCt);  
    }
    else {
      // 1 "nums[i]" seen. all smaller numbers seen twice or more
      elementDetails[1] = 1; 

      subsetMap.put(nums[i], elementDetails);  
    }   
    continue; 
  }
  ```
- Instead, it should be placed like this:
  ```java
  if(countOfSqrt >= 1) {
    subsetMap.remove(sqrt);
    if(countOfSqrt == 1) {
      // Since sqrt only seen 1 time, subset ends here.
      maxCt = Math.max(getCount(sqrt, elementDetails), maxCt);  
    }
    else {
      // 1 "nums[i]" seen. all smaller numbers seen twice or more
      elementDetails[1] = 1; 

      subsetMap.put(nums[i], elementDetails);  
      
      continue; 
    }   
  }
  ```
### Screenshots

![image](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/67020572/50e67037-80db-480a-b2e8-7db76df6b519)

### Additional context

Successful submission of the erroneous code: https://leetcode.com/problems/find-the-maximum-number-of-elements-in-subset/submissions/1158956850.

**Comment by LC-Pam** [comment]
Hi there,

Thank you for bringing your feedback to our attention. We've conducted a thorough investigation into the issue you raised, and after careful examination, our findings affirm that our judge is working correctly.


<img width="303" alt="Screen Shot 2024-01-29 at 11 18 39 AM" src="https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/103138242/2e578e98-002c-439a-9c2c-9636b3f8f390">

We appreciate the time and effort you've invested in sharing your feedback. It's invaluable to us as we work towards maintaining a high-quality platform.

Feel free to continue the dialogue here or provide any additional insights. We value your input and strive to create a collaborative space for improvement.

Best regards,
The LeetCode Team

**Comment by rohan-v8rma** [comment]
I was not suggesting a problem being present with the judge in the first place.

The test case I was talking about was `nums=[1,2,4,4,16]`, as shown in the attached screenshot. The length of the subset `[4, 16, 4]` of the `nums` array, which is 3, was the answer.

When this test case was manually added in the test cases section by me, my code failed to pass, however I was able to submit the code without any errors.

But no worries, I tried submitting my erroneous code again right now, and other test cases have been added preventing its submission.
